AutomatedTesting/Gem/PythonTests/scripting/NodeCategory_ExpandOnClick.py

```diff
@@ -45,79 +52,87 @@ def NodeCategory_ExpandOnClick():
-        for i in range(5):  # this range can be increased for safe side
-            QtTest.QTest.mouseClick(
-                item_view.viewport(),
-                QtCore.Qt.LeftButton,
-                QtCore.Qt.NoModifier,
-                QtCore.QPoint(rect_left_x - i, rect_center_y),
-            )
-            if item_view.isExpanded(index) != original_state:
-                break
-
-    def double_click(item_view, index):
-        item_index_center = item_view.visualRect(index).center()
+
+    def __init__(self):
+        self.editor_main_window = None
+        self.sc_editor = None
+        self.sc_editor_main_window = None
+
+
+    def left_click_expander_button(self, node_palette_node_tree, category_index):
+        rect_left_x = 1 # 1 pixel from the left side of the widget looks like where the expand button begins
+        rect_center_y = node_palette_node_tree.visualRect(category_index).center().y()
+        click_position = QtCore.QPoint(rect_left_x, rect_center_y)
+        # click position relative to node palette tree view and not screen space xy
+        QtTest.QTest.mouseClick(node_palette_node_tree.viewport(),
+                                QtCore.Qt.LeftButton,
+                                QtCore.Qt.NoModifier,
+                                click_position,
+                                )
+
+    def double_click_node_category(self, node_palette_node_tree, index):
+        item_index_center = node_palette_node_tree.visualRect(index).center()
         # Left click on the item before trying to double click, will otherwise fail to expand
         # as first click would highlight and second click would be a 'single click'
-        pyside_utils.item_view_index_mouse_click(item_view, index)
-        QTest.mouseDClick(item_view.viewport(), QtCore.Qt.LeftButton, QtCore.Qt.NoModifier, item_index_center)
+        pyside_utils.item_view_index_mouse_click(node_palette_node_tree, index)
+        QtTest.QTest.mouseDClick(node_palette_node_tree.viewport(),
+                                 QtCore.Qt.LeftButton,
+                                 QtCore.Qt.NoModifier,
+                                 item_index_center)
+
+    def wait_and_verify_category_expanded(self, test_case, node_palette_node_tree, node_palette_math_category):
+        category_has_expanded = helper.wait_for_condition(
+            lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
+        Report.result(test_case, category_has_expanded)
 
-    # 1) Open Script Canvas pane
-    general.open_pane("Script Canvas")
-    Report.critical_result(Tests.pane_open, general.is_pane_visible("Script Canvas"))
+    def wait_and_verify_category_collapsed(self, test_case, node_palette_node_tree, node_palette_math_category):
+        category_has_expanded = helper.wait_for_condition(
+            lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
+        Report.result(test_case, not category_has_expanded)
```

Comment:
These are the same function, but returning flipped result. You can handle this conditionally in the function to reduce the code even further with something like the following: 


```
def verify_category_expander_state(self, test_case, node_palette_node_tree, node_pallette_math_category, expanded = True)
        category_has_expanded = helper.wait_for_condition(
            lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
        Report.result(test_case, category_has_expanded == expanded)
```